### PR TITLE
Merge main branch into dev-1.29

### PR DESCRIPTION
**Pull request by katcosgrove**
Merging main into dev-1.29 to maintain branch sync

### CODE REVIEW

**Comment by reylejano** [comment]
clean sync
/approve

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -533,25 +548,36 @@ Each feature gate is designed for enabling/disabling a specific feature:
   and volume controllers.
 - `InTreePluginvSphereUnregister`: Stops registering the vSphere in-tree plugin in kubelet
   and volume controllers.
+<<<<<<< HEAD
+=======
+- `JobMutableNodeSchedulingDirectives`: Allows updating node scheduling directives in
+  the pod template of [Job](/docs/concepts/workloads/controllers/job/).
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:554]: Doesn't look right.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -507,6 +517,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
   for more details.
 - `GracefulNodeShutdownBasedOnPodPriority`: Enables the kubelet to check Pod priorities
   when shutting down a node gracefully.
+<<<<<<< HEAD
+=======
+- `GRPCContainerProbe`: Enables the gRPC probe method for liveness, readiness and startup probes.
+  See [Configure Liveness, Readiness and Startup Probes](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe).
+>>>>>>> upstream/main
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:524]: Doesn't look right.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -563,28 +589,35 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
   When enabled, kubelet CRI interface and authenticated http servers are instrumented to generate
   OpenTelemetry trace spans.
+<<<<<<< HEAD
   See [Traces for Kubernetes System Components](/docs/concepts/cluster-administration/system-traces) for more details.
+=======
+  See [Traces for Kubernetes System Components](/docs/concepts/cluster-administration/system-traces/)
+  for more details.
+- `LegacyServiceAccountTokenNoAutoGeneration`: Stop auto-generation of Secret-based
+  [service account tokens](/docs/concepts/security/service-accounts/#get-a-token).
+>>>>>>> upstream/main
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:620]: Doesn't look right.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -637,31 +670,42 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
   The field lets you expose host IP addresses to workloads.
+<<<<<<< HEAD
 - `PodIndexLabel`: Enables the Job controller and StatefulSet controller to add the pod index as a label when creating new pods. See [Job completion mode docs](/docs/concepts/workloads/controllers/job#completion-mode) and [StatefulSet pod index label docs](/docs/concepts/workloads/controllers/statefulset/#pod-index-label) for more details.
 - `PodLifecycleSleepAction`: Enables the `sleep` action in Container lifecycle hooks.
+=======
+- `PodIndexLabel`: Enables the Job controller and StatefulSet controller to add the pod index as a label
+  when creating new pods. See [Job completion mode docs](/docs/concepts/workloads/controllers/job/#completion-mode)
+  and [StatefulSet pod index label docs](/docs/concepts/workloads/controllers/statefulset/#pod-index-label)
+  for more details.
+>>>>>>> upstream/main
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:704]: Doesn't look right.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 5]
```diff
@@ -726,9 +770,17 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
   in EndpointSlices. See [Topology Aware
   Hints](/docs/concepts/services-networking/topology-aware-hints/) for more
   details.
+=======
+  in EndpointSlices. See [Topology Aware Hints](/docs/concepts/services-networking/topology-aware-hints/)
+  for more details.
+- `TopologyManager`: Enable a mechanism to coordinate fine-grained hardware resource
+  assignments for different components in Kubernetes. See
+  [Control Topology Management Policies on a node](/docs/tasks/administer-cluster/topology-manager/).
+>>>>>>> upstream/main
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:783]: Doesn't look right.

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks

/hold

Please remove the merge conflicts (squashing them out), and force-push. Or make an equivalent change.

**Comment by sftim** [comment]
/approve cancel

with https://github.com/kubernetes/website/pull/43908#pullrequestreview-1729317170 in mind. OK to re-approve once this has an appropriate LGTM.

**Comment by sftim** [comment]
@katcosgrove could you squash the latest commit into its (merge commit) parent? Let us know if you've a concern around doing this.

Reason: the feature gates file is one we try to be extra nice about as merge conflicts there are a frustration (which you've kinda seen).

**Comment by katcosgrove** [comment]
Closing this and re-doing due to git being haunted and that being faster than unraveling the problem

**Pull request closed without merging**